common/djangoapps/enrollment/views.py

```diff
@@ -469,6 +472,14 @@ def get(self, request):
                     ).format(username=username)
                 }
             )
+        if username == request.user.username or self.has_api_key_permissions(request):
+            return Response(enrollment_data)
+        filtered_data = []
+        for enrollment in enrollment_data:
```

Comment:
actually one good-looking function is here: https://github.com/edx/edx-platform/blob/dsjen/courses-via-overview/lms/djangoapps/oauth2_handler/handlers.py#L169